noir_stdlib/src/hash.nr

```diff
@@ -46,8 +51,25 @@ pub fn pedersen_hash<N>(input: [Field; N]) -> Field
     pedersen_hash_with_separator(input, 0)
 }
 
-#[foreign(pedersen_hash)]
-pub fn pedersen_hash_with_separator<N>(input: [Field; N], separator: u32) -> Field {}
+#[field(bn254)]
+fn derive_generators<N, M>(domain_separator_bytes: [u8; M], starting_index: u32) -> [EmbeddedCurvePoint; N] {
+    crate::assert_constant(domain_separator_bytes);
+    crate::assert_constant(starting_index);
+    __derive_generators(domain_separator_bytes, starting_index)
+}
+
+#[builtin(derive_pedersen_generators)]
```

Comment:
Do we also need to add support for this on the AVM side? Not sure how builtins work. @sirasistant

Reply:
Nah, builtins are all applied by the compiler so this call will be replaced with an array literal containing the pedersen generators.

Reply:
Awesome thanks